src/app/active-mode/Views/current-activity/VendorBounties.tsx

```diff
@@ -1,110 +1,179 @@
+    vendorsResponse: state.vendors.vendorsByCharacter[props.store.id]?.vendorsResponse,
   };
 }
 
-type Props = ProvidedProps & StoreProps;
+function bountiesForActivity(
+  defs: D2ManifestDefinitions,
+  bounties: VendorItem[],
+  activityInfo: DestinyCharacterActivitiesComponent
+) {
+  const activity = defs.Activity.get(activityInfo.currentActivityHash);
+  const activityMode = defs.ActivityMode[activityInfo.currentActivityModeHash];
+
+  return bounties.filter(({ item }) => {
+    const info = item?.hash && pursuitsInfo[item?.hash];
+
+    if (!info) {
+      // Until the pursuits.json matches all bounties, we need to return falsy
+      return false;
+    }
+
+    if (!info?.ActivityMode && !info?.Destination) {
+      // Show all bounties that can be completed anywhere
+      return true;
+    }
+
+    const matchingDestination = info.Destination?.includes(activity?.destinationHash);
+    const matchingActivity =
+      (!info.Destination && info.ActivityMode?.includes(activityMode?.hash)) ||
+      activityMode?.parentHashes.some((hash) => info.ActivityMode?.includes(hash));
+
+    return matchingDestination || matchingActivity;
+  });
+}
 
-/** Find unclaimed vendor bounties based on your current activity */
+/** Find relevant vendor bounties based on your current activity */
 function VendorBounties({
   defs,
-  vendors,
+  vendorsResponse,
   store,
-  activity,
+  activityInfo,
   buckets,
   ownedItemHashes,
   account,
 }: Props) {
-  if (!vendors) {
-    return null;
-  }
-
-  const bounties: VendorItem[] = [];
-  const vendorData = store.id ? vendors[store.id] : undefined;
-  const vendorsResponse = vendorData?.vendorsResponse;
-  const vendorHashes: Vendors[] = [];
-  activity.activityModeTypes?.forEach((modeType) => {
-    const vendors = vendorsByActivityModeType[modeType];
-    vendors && vendorHashes.push(...vendors);
-  });
+  const [suggestedBounties, setSuggestedBounties] = useState<VendorItem[]>([]);
+  const [bounties, setBounties] = useState<VendorItem[]>([]);
 
-  if (!vendorHashes.length) {
-    const vendors = vendorsByDestinationHash[activity.placeHash];
-    vendors && vendorHashes.push(...vendors);
-  }
-
-  vendorHashes.forEach((vendorHash) => {
-    const vendor = vendorsResponse?.vendors.data?.[vendorHash];
-    const d2Vendor = toVendor(
-      vendorHash,
-      defs,
-      buckets,
-      vendor,
-      account,
-      vendorsResponse?.itemComponents[vendorHash],
-      vendorsResponse?.sales.data?.[vendorHash]?.saleItems,
-      {}
-    );
-    const vendorBounties = d2Vendor?.items.filter(
-      ({ item, canPurchase, canBeSold }: VendorItem) =>
-        canPurchase && canBeSold && item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties)
-    );
+  useEffect(() => {
+    if (!vendorsResponse) {
+      return;
+    }
+
+    const purchasableBounties: VendorItem[] = [];
+    const { vendors, sales, itemComponents } = vendorsResponse;
+
+    Object.values(vendors.data || []).forEach((vendor) => {
```

Comment:
Every place we use a `forEach` is probably better done as `for (x of y)`

Reply:
updated (moved to activity-util.ts)